### PR TITLE
Update NLSProblems for NLPModelJuMP 0.6

### PR DESCRIPTION
**Pull request by amontoison**
Update the package for NLPModelsJuMP 0.6  / JuMP 0.21 and NLPModel 0.12.

### CODE REVIEW

File: src/hs13.jl [hunk 1]
```diff
@@ -15,10 +15,10 @@ export hs13
 function hs13(args...)
 
   model = Model()
-  @variable(model, x[1:2] >= 0.0, start=-2.0)
+  @variable(model, x[1:2] ≥ 0.0, start=-2.0)
```
Comment (dpo) [inline src/hs13.jl]: It's a good idea to replace those, but there are more that you didn't change. I think they should be changed in another PR as it's unrelated to the other changes.

File: src/tp268.jl [hunk 2]
```diff
@@ -33,5 +33,5 @@ function tp268(args...)
   @constraint(nls,  8 * x[1] -      x[2] + 2 * x[3] + 5 * x[4] - 3 * x[5] - 11 ≥ 0)
   @constraint(nls, -4 * x[1] -  2 * x[2] + 3 * x[3] - 5 * x[4] +     x[5] + 30 ≥ 0)
```
Comment (dpo) [inline src/tp268.jl]: Do these have to become `@NLconstraint` or not? Ideally, it would be better to detect linear constraints properly.

Comment (amontoison) [reply src/tp268.jl]: I didn't modified them in `OptimizationProblems.jl`, I think we should keep @constraint. We will detect linear constraints properly in the future with `NLPModelsJuMP`, one step after another.

File: src/hs13.jl (hunk 1 above)
Comment (amontoison) [reply src/hs13.jl]: I modified some of them because they were near the `setvalue` that I changed, but I only did a few files.
I remembered that I can use a sed call for that... ...in an other pull request.

Comment (dpo) [reply src/hs13.jl]: Let's keep PRs focused.

**Comment by amontoison** [comment]
One model (`mgh35`) can't be updated with JuMP ≥ 0.19. Should I remove it?
https://discourse.julialang.org/t/nlexpression-with-jump-0-21/34859/2

**Comment by abelsiqueira** [comment]
I'm fine with removing it.

**Comment by amontoison** [comment]
Thanks Abel, I removed `mgh35` problem.

**Comment by dpo** [comment]
Does it mean it's impossible to implement that problem with JuMP?

**Comment by amontoison** [comment]
We can implement that problem with JuMP but we cannot use NLexpression. We need to create the expression ourself and give the `Expr` to `set_NL_objective` and `add_NL_constraint`.

In our specific case, we explicitly need the NLexpression for building an `MathOptNLSModel` because we use them for the constraints of the Fmodel and the objective of the cmodel.

**Comment by amontoison** [comment]
It's a pity...

**Comment by amontoison** [comment]
Close / reopen this pull request when https://github.com/JuliaSmoothOptimizers/NLPModelsJuMP.jl/pull/38 is merged.

**Comment by dpo** [comment]
Thank you.

**Comment by dpo** [comment]
Note for the future: that's not the right way to trigger builds!
